mlflow/sklearn/__init__.py

```diff
@@ -945,6 +1052,58 @@ def patched_fit(original, self, *args, **kwargs):
                     FLAVOR_NAME, class_def, func_name, patched_fit, manage_run=True,
                 )
 
+        for func_name in ["predict", "transform", "predict_proba"]:
+            if hasattr(class_def, func_name):
+                safe_patch(
+                    FLAVOR_NAME, class_def, func_name, patched_predict, manage_run=False,
+                )
+
+    from sklearn import metrics
+    # TODO: patch all metrics API.
+    #   includes all methods ends with _score/_error/_loss in `sklearn.metrics`
+    #   but there're few metric methods ends with other words, like mean_poisson_deviance
+
+    # Note:
+    #  don't need patch model.score() because internally model.score() will call the concrete metric API
+    for metric_method_name in metrics.__all__:
+        # excludes '**Display' object such as `ConfusionMatrixDisplay`
+        if callable(getattr(metrics, metric_method_name)) and not metric_method_name.endswith('Display'):
+            safe_patch(FLAVOR_NAME, metrics, metric_method_name, patched_metric_api, manage_run=False)
+
+    def patched_scorer_call(original, self, *args, **kwargs):
+        global _autolog_training_status
+
+        _autolog_training_status.in_scorer_call_scope = True
+        metric = original(self, *args, **kwargs)
+        _autolog_training_status.in_scorer_call_scope = False
+
+        if not _autolog_training_status.in_fit_call_scope:
+            estimator = args[0]
+            if id(estimator) == _autolog_training_status.model_id:
+                # TODO: refine metric_name to make it include arguments set for the metric
+                metric_name = self._score_func.__name__
+                with ResumeAutologRun():
+                    log_metric_index = _autolog_training_status.get_log_test_metric_index(metric_name)
+                    mlflow.log_metric(
+                        f'test_{metric_name}_{log_metric_index}',
+                        metric
+                    )
+
+        return metric
+
+    def patched_make_scorer(original, *args, **kwargs):
+        # TODO:
+        #  check _SklearnTrainingSession and skip logging if in nested runs.
+        global _autolog_training_status
+        scorer = original(*args, **kwargs)
+        safe_patch(
+            FLAVOR_NAME, scorer, '__call__', patched_scorer_call, manage_run=False,
+        )
+        return scorer
+
+    # Patch metrics.make_scorer
+    safe_patch(FLAVOR_NAME, metrics, 'make_scorer', patched_make_scorer, manage_run=False)
```

Comment:
This is for patching for this case:
https://scikit-learn.org/stable/modules/model_evaluation.html#implementing-your-own-scoring-object

example:
```
from sklearn.metrics import fbeta_score, make_scorer
ftwo_scorer = make_scorer(fbeta_score, beta=2, average='weighted')

ftwo_scorer(classifier_model, test_iris_X, test_iris_y)  # then it will autolog test_fbeta_score metric
```